DuckDuckGo/HomePage/Model/HomePageAddressBarModel.swift

```diff
@@ -87,15 +87,29 @@ extension HomePage.Models {
 
         let tabCollectionViewModel: TabCollectionViewModel
 
+        private var isExperimentActive: Bool = false {
+            didSet {
+                if isExperimentActive {
+                    let ntpExperiment = NewTabPageSearchBoxExperiment()
+                    ntpExperiment.assignUserToCohort()
+                    shouldShowAddressBar = ntpExperiment.cohort?.isExperiment == true
+                }
+            }
+        }
         private var privacyConfigCancellable: AnyCancellable?
         private var addressBarViewControllerCancellables = Set<AnyCancellable>()
 
         init(tabCollectionViewModel: TabCollectionViewModel, privacyConfigurationManager: PrivacyConfigurationManaging) {
             self.tabCollectionViewModel = tabCollectionViewModel
-            self.shouldShowAddressBar = privacyConfigurationManager.privacyConfig.isEnabled(featureKey: .newTabSearchField)
+            self.isExperimentActive = privacyConfigurationManager.privacyConfig.isEnabled(featureKey: .newTabSearchField)
+            if self.isExperimentActive {
+                let ntpExperiment = NewTabPageSearchBoxExperiment()
+                ntpExperiment.assignUserToCohort()
+                self.shouldShowAddressBar = ntpExperiment.cohort?.isExperiment == true
```

Comment:
didSet doesn't get called on the initial value assignment unfortunately. But I'll extract this code into a helper function to avoid code repetition 👍